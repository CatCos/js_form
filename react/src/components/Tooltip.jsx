import React from 'react';
import PropTypes from 'prop-types';

/**
 * Render Tooltip component
 *
 * @params props Component properties
 */
function Tooltip(props) {

  return (
    <div className="Tooltip Tooltip-error" id="tooltip">
      <div className="Tooltip__arrow"></div>
      <div className="Tooltip__balloon">
        <span className="Tooltip__label" id="tooltip-span">{props.value}</span>
      </div>
    </div>
  );
}

Tooltip.propTypes = {
  value: PropTypes.string
};

Tooltip.defaultProps = {
  value: ''
};

export default Tooltip;
