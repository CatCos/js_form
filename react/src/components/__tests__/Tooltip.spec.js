import React from 'react';
import Tooltip from '../Tooltip';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Tooltip
      value="Required"/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
