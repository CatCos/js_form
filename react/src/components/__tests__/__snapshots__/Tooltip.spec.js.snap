// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Tooltip Tooltip-error"
  id="tooltip"
>
  <div
    className="Tooltip__arrow"
  />
  <div
    className="Tooltip__balloon"
  >
    <span
      className="Tooltip__label"
      id="tooltip-span"
    >
      Required
    </span>
  </div>
</div>
`;
